theano/gpuarray/tests/test_GpuMaxAndArgmax.py

```diff
@@ -0,0 +1,139 @@
+from unittest import TestCase
+from theano.tests import unittest_tools as utt
+import numpy as np
+import theano
+import theano.tensor as T
+
+
+def randomTensor(*shapes):
+    dimlist = shapes
+    size = 1
+    for dimsize in dimlist:
+        size *= dimsize
+    return np.random.normal(size=size).astype(np.float32).reshape(dimlist)
+
+
+def numpyMaxAndArgmax(X, axis=None):
+    if axis is None:
+        axis = range(X.ndim)
+    elif not isinstance(axis, (tuple, list)):
+        axis = [axis]
+    axis = list(set(axis))  # remove duplicated values.
+    axis.sort()
+    axis = tuple(axis)
+    ref_max = np.max(X, axis=axis)
+    # Numpy does not support multiple axes for argmax. Work around
+    # Code copied from MaxAndArgmax.perform()
+    keep_axes = np.array([i for i in range(X.ndim) if i not in axis], dtype='int64')
+    # Not-reduced axes in front
+    transposed_x = np.transpose(X, np.concatenate((keep_axes, axis)))
+    kept_shape = transposed_x.shape[:len(keep_axes)]
+    reduced_shape = transposed_x.shape[len(keep_axes):]
+    new_shape = kept_shape + (np.prod(reduced_shape),)
+    reshaped_x = transposed_x.reshape(new_shape)
+    return (ref_max, np.argmax(reshaped_x, axis=-1))
+
+
+class TestGpuMaxAndArgmax(TestCase):
+    # We run all tests with 5-D tensors of 10 000 000 elements.
+    # NB: In each test, any first call of theano function should be ignored
+    # with Theano config flag profiling.ignore_first_call=True.
+    # To just check if GpuMaxAndArgmax is called:
+    # $ theano-cache purge && THEANO_FLAGS=floatX=float32,device=cuda,profile=True,profiling.ignore_first_call=True \
```

Comment:
Ok! As I wrote and modified C code frequently on the issues I have worked, I used to always clear the cache so that any C code (or compiled code ?) is reset before running the tests.

From now I will clear the cache only when I indeed modify a C code (or is it useless) ?